sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java

```diff
@@ -35,85 +42,99 @@
   private static final String URN_BEAM_LOGICAL_DECIMAL = "beam:logical_type:decimal:v1";
   private static final String URN_BEAM_LOGICAL_JAVASDK = "beam:logical_type:javasdk:v1";
 
-  public static SchemaApi.Schema schemaToProto(Schema schema) {
+  public static SchemaApi.Schema schemaToProto(Schema schema, boolean serializeLogicalType) {
```

Comment:
This is a temporary hack to make OneOf comparison work, since right now the logical type serialization uses Java Serialization which makes the protos not binary comparable. Once we have a better serialization of logical types this can go away. Also once we support recursive schemas (in progress), then OneOf no longer needs to use the proto as its arg.